From cda39c22755275185dfb601e05c1409c20abf2ac Mon Sep 17 00:00:00 2001
From: Ariel Aviad <aaviad@habana.ai>
Date: Wed, 14 May 2025 14:36:41 +0300
Subject: [PATCH] [SW-228312] g3: bring up: update files to match habanalabs repo

Habanalabs repo update gaudi3 bring up, this
patch update the changes to embedded specs

Change-Id: I1c8a0c007730408e5ed8c1c6413425fe2571674e
Signed-off-by: Ariel Aviad <aaviad@habana.ai>
---

diff --git a/gaudi3/bringup/gaudi3_bringup.c b/gaudi3/bringup/gaudi3_bringup.c
index a5414d5..4ce6356 100644
--- a/gaudi3/bringup/gaudi3_bringup.c
+++ b/gaudi3/bringup/gaudi3_bringup.c
@@ -449,19 +449,6 @@
 	SPI_INTR_DEC1_TRACE_N_DBG,
 };
 
-enum eco_bfe {
-	NIC_ENABLE_H9_RX_DROP_ECO = 0,
-	NIC_ENABLE_H9_QP_DOORBELLS_ECO,
-	NIC_ENABLE_H9_CC_MSG_DROPS_ECO,
-	NIC_ENABLE_H9_REMOTE_PI_UPDATE_ECO,
-	NIC_ENABLE_H9_RXB_MEM_DEADLOCK_ECO,
-	NIC_ENABLE_H9_SINGLE_QP_PERF_FIX_ECO,
-	NIC_ENABLE_H9_SAL_OVERRIDE_ECO,
-	NIC_ENABLE_H9_SACK_DEADLOCK_ECO,
-	NIC_ENABLE_H9_TXE_BUFF_ALLOC_ECO,
-	NIC_ENABLE_H9_PHY_MAC_HANG_ECO,
-};
-
 #define SPI_INTR_STLB_BASE SPI_INTR_STLB_FAULT
 
 typedef void (*shared_aggr_handle_and_clear)(struct hl_device *hdev, u32 die,
@@ -631,23 +618,11 @@
 	[GAUDI3_D0_G2_NIC_PLL] = mmD0_G2_NIC_PLL_CTRL_BASE,
 };
 
-struct gaudi3_etr_ac_config gaudi3_etr_ac_config[GAUDI3_NUM_ETR] = {
-	[GAUDI3_D0_PSOC_ETR] = {
-		mmD0_PSOC_ETR_AC_BASE - mmD0_NCH_AC_BASE,
-		mmD0_PSOC_ETR_BASE - mmD0_NCH_ETR_BASE
-		},
-	[GAUDI3_D0_NCH_ETR] = {
-		mmD0_NCH_AC_BASE - mmD0_NCH_AC_BASE,
-		mmD0_NCH_ETR_BASE - mmD0_NCH_ETR_BASE
-		},
-	[GAUDI3_D1_PSOC_ETR] = {
-		mmD1_PSOC_ETR_AC_BASE - mmD0_NCH_AC_BASE,
-		mmD1_PSOC_ETR_BASE - mmD0_NCH_ETR_BASE
-		},
-	[GAUDI3_D1_NCH_ETR] = {
-		mmD1_NCH_AC_BASE - mmD0_NCH_AC_BASE,
-		mmD1_NCH_ETR_BASE - mmD0_NCH_ETR_BASE
-		},
+static const u64 gaudi3_etr_ac_base[GAUDI3_NUM_ETR] = {
+	[GAUDI3_D0_PSOC_ETR] = mmD0_PSOC_ETR_AC_BASE,
+	[GAUDI3_D0_NCH_ETR]  = mmD0_NCH_AC_BASE,
+	[GAUDI3_D1_PSOC_ETR] = mmD1_PSOC_ETR_AC_BASE,
+	[GAUDI3_D1_NCH_ETR]  = mmD1_NCH_AC_BASE,
 };
 
 #define GAUDI3_INIT_PLL_COEFFICIENT(pll, refdiv, fbdiv, pdiv1, pdiv2) \
@@ -818,36 +793,6 @@
 				ROTATOR_COORD_NUM_ERR_CONTEXT_ID_COORD_M}
 };
 
-bool gaudi3_get_bfe_status(struct hbl_aux_dev *aux_dev, u8 bfe)
-{
-	struct hl_cn *cn = container_of(aux_dev, struct hl_cn, cn_aux_dev);
-	struct hl_device *hdev = container_of(cn, struct hl_device, cn);
-
-	switch (bfe) {
-	case NIC_ENABLE_H9_RX_DROP_ECO:
-		return hdev->nic_enable_h9_rx_drop_eco;
-	case NIC_ENABLE_H9_QP_DOORBELLS_ECO:
-		return hdev->nic_enable_h9_qp_doorbells_eco;
-	case NIC_ENABLE_H9_CC_MSG_DROPS_ECO:
-		return hdev->nic_enable_h9_cc_msg_drops_eco;
-	case NIC_ENABLE_H9_REMOTE_PI_UPDATE_ECO:
-		return hdev->nic_enable_h9_remote_pi_update_eco;
-	case NIC_ENABLE_H9_RXB_MEM_DEADLOCK_ECO:
-		return hdev->nic_enable_h9_rxb_mem_deadlock_eco;
-	case NIC_ENABLE_H9_SINGLE_QP_PERF_FIX_ECO:
-		return hdev->nic_enable_h9_single_qp_perf_fix_eco;
-	case NIC_ENABLE_H9_SAL_OVERRIDE_ECO:
-		return hdev->nic_enable_h9_sal_override_eco;
-	case NIC_ENABLE_H9_SACK_DEADLOCK_ECO:
-		return hdev->nic_enable_h9_sack_deadlock_eco;
-	case NIC_ENABLE_H9_TXE_BUFF_ALLOC_ECO:
-		return hdev->nic_enable_h9_txe_buff_alloc_eco;
-	case NIC_ENABLE_H9_PHY_MAC_HANG_ECO:
-		return hdev->nic_enable_h9_phy_mac_hang_eco;
-	}
-
-	return false;
-}
 
 /**
  * gaudi3_get_spi_gw_addr - convert address to SPI GW address
@@ -2963,29 +2908,25 @@
 				      u8 pos, enum gaudi3_ac_instruction opcode,
 				      u32 addr, u32 data)
 {
-	u64 ac_off = gaudi3_etr_ac_config[etr_idx].ac_off;
+	u64 ac_off = gaudi3_etr_ac_base[etr_idx];
 
-	WREG32(mmD0_NCH_AC_BASE + mmAUTONOMOUS_CONTROL_CMD_ENTRY_0 + ac_off + pos * sizeof(u32),
+	WREG32(mmAUTONOMOUS_CONTROL_CMD_ENTRY_0 + ac_off + pos * sizeof(u32),
 			opcode);
-	WREG32(mmD0_NCH_AC_BASE + mmAUTONOMOUS_CONTROL_ADDR_ENTRY_0 + ac_off + pos * sizeof(u32),
+	WREG32(mmAUTONOMOUS_CONTROL_ADDR_ENTRY_0 + ac_off + pos * sizeof(u32),
 			addr);
-	WREG32(mmD0_NCH_AC_BASE + mmAUTONOMOUS_CONTROL_DATA_ENTRY_0 + ac_off + pos * sizeof(u32),
+	WREG32(mmAUTONOMOUS_CONTROL_DATA_ENTRY_0 + ac_off + pos * sizeof(u32),
 			data);
 }
 
 static void gaudi3_ac_program(struct hl_device *hdev, u32 etr_idx, u32 buf_size)
 {
+	u64 ac_off = gaudi3_etr_ac_base[etr_idx];
 	u32 doorbell, poll_mask, rwp, rwp_msb_mask;
-	u64 ac_off, etr_off;
 	u8 i = 0;
 
-	ac_off = gaudi3_etr_ac_config[etr_idx].ac_off;
-	etr_off = gaudi3_etr_ac_config[etr_idx].etr_off;
-
 	doorbell = CFG_BAR_BASE - LBW_BASE + mmD0_PCIE_DBI_SIG_BASE + mmPCIE_DBI_MSIX_DOORBELL_OFF;
-	poll_mask = CFG_BAR_BASE - LBW_BASE + mmD0_NCH_AC_BASE +
-			mmAUTONOMOUS_CONTROL_POLLING_MASK + ac_off;
-	rwp = CFG_BAR_BASE - LBW_BASE + mmD0_NCH_ETR_BASE + mmETR_RWP + etr_off;
+	poll_mask = CFG_BAR_BASE - LBW_BASE + mmAUTONOMOUS_CONTROL_POLLING_MASK + ac_off;
+	rwp = CFG_BAR_BASE - LBW_BASE + mmETR_RWP + gaudi3_etr_base[etr_idx];
 
 	/* This assumes etr_buf_dram_size is power of 2 */
 	rwp_msb_mask = buf_size >> 1;
@@ -3019,26 +2960,22 @@
 	/* At this point, AC will go back to the first instruction */
 
 	/* Modify the number of commands, without the activation bit */
-	RMWREG32(mmD0_NCH_AC_BASE + mmAUTONOMOUS_CONTROL_CTRL + ac_off, i - 1,
+	RMWREG32(ac_off + mmAUTONOMOUS_CONTROL_CTRL, i - 1,
 				AUTONOMOUS_CONTROL_CTRL_NUM_CMD_M);
 }
 
 void gaudi3_ac_start_no_fw(struct hl_device *hdev, u32 etr_idx, u32 buf_size)
 {
-	u64 base = gaudi3_etr_ac_config[etr_idx].ac_off;
-
 	/* program ac before enable */
 	gaudi3_ac_program(hdev, etr_idx, buf_size);
 
-	RMWREG32(mmD0_NCH_AC_BASE + mmAUTONOMOUS_CONTROL_CTRL + base, 1,
+	RMWREG32(gaudi3_etr_ac_base[etr_idx] + mmAUTONOMOUS_CONTROL_CTRL, 1,
 			AUTONOMOUS_CONTROL_CTRL_EN_M);
 }
 
 void gaudi3_ac_stop_no_fw(struct hl_device *hdev, u32 etr_idx)
 {
-	u64 base = gaudi3_etr_ac_config[etr_idx].ac_off;
-
-	RMWREG32(mmD0_NCH_AC_BASE + mmAUTONOMOUS_CONTROL_CTRL + base, 0,
+	RMWREG32(gaudi3_etr_ac_base[etr_idx] + mmAUTONOMOUS_CONTROL_CTRL, 0,
 			AUTONOMOUS_CONTROL_CTRL_EN_M);
 }
 
diff --git a/gaudi3/bringup/hbl_cn/gaudi3_cn_bringup.c b/gaudi3/bringup/hbl_cn/gaudi3_cn_bringup.c
index 1460bfb..200f97b 100644
--- a/gaudi3/bringup/hbl_cn/gaudi3_cn_bringup.c
+++ b/gaudi3/bringup/hbl_cn/gaudi3_cn_bringup.c
@@ -947,7 +947,9 @@
 					  BIT(NIC_ACCUMULATE_FEC_DURATION) |
 					  BIT(NIC_PRINT_FEC_STATS) |
 					  BIT(NIC_TRIGGER_LINK_SHUTDOWN) |
-					  BIT(NIC_INVALIDATE_PORT_QPS);
+					  BIT(NIC_INVALIDATE_PORT_QPS) |
+					  BIT(NIC_QP_ID_OFFSET_IN_MACRO) |
+					  BIT(NIC_FORCE_PORT_FAILURE);
 
 	hdev->debugfs_supp_mask |= gaudi3_debugfs_support_mask;
 }
